[DanzaSeah] iP

DanzaSeah opened the pull request:
# BabbleBot
Hey, hey! It’s me, BabbleBot! I'm here to keep you on top of everything you need to do--But fair warning, if something doesn’t quite make sense to me, well, you might just hear me babble about it!

## Some interesting aspects of BabbleBot are that it's:
- Text-based
- Data persistent
- Babbles!

## Here's how you can get started using BabbleBot:
1. Download it from [here](https://github.com/DanzaSeah/ip/releases/tag/A-Jar)!
2. Place the downloaded jar file in a directory of your preference
3. Open cmd and change directory to the location of the file with `cd C:\path\to\your\jar`
4. Run `java -jar babblebot.jar`
5. Start adding tasks!

Oh, and did I forget to mention, this is all for **FREE**!

## Here's some features of BabbleBot:
- [x] Managing Todos
- [x] Managing Deadlines
- [x] Managing Events
- [ ] Babbling! 

I'm sure you don't want me to babble on and on about an invalid input, so my best advice to you is to ***tread lightly*** 😠 
``` java
   public void showBabbleBotError() {
        System.out.println(SEPARATOR);
        System.out.println("Whoopsie daisy! Looks like I got all tangled up in my words there!\n" +
                "Let's try that again in a way that might make a bit more sense.\n" +
                "What do you need help with?");
        System.out.println(SEPARATOR);
    }
```

> “Say the words. Say you want this. Nothing happens until I hear you say it.” - Jesse Pinkman ([source](https://youtu.be/dfZgkWlb_hw?si=Euo1kiO-tZ4YYKIo&t=104))

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

DanzaSeah committed:
Rename main class

DanzaSeah committed:
Add greetings

DanzaSeah committed:
Add support for echo

DanzaSeah committed:
Added support for list (Can add items to a list and display them)

DanzaSeah committed:
Added new Task class

DanzaSeah committed:
Add ability to mark task as done

DanzaSeah committed:
Add ability to unmark tasks to become undone

DanzaSeah committed:
Added toString() method in the class "Task"

DanzaSeah committed:
Created new classes Deadline, Event and Todo

DanzaSeah committed:
Add support for tracking three types of tasks, i.e Todo, Deadline and Event

DanzaSeah committed:
Implemented Text UI testing

DanzaSeah committed:
Created new BabbleBotException class

DanzaSeah committed:
Added Exception handling capabilities for 2 basic cases

DanzaSeah committed:
Add support for deleting tasks from the list

DanzaSeah committed:
Add toFileFormat() method to the four task classes

DanzaSeah committed:
Modify return statement

DanzaSeah committed:
Add support to save and load tasks

DanzaSeah committed:
Merge branch 'branch-Level-7'

DanzaSeah committed:
Add support for date comprehension and formatting

DanzaSeah committed:
Merge branch 'branch-Level-8'

DanzaSeah committed:
Add Ui class

DanzaSeah committed:
Add Storage class

DanzaSeah committed:
Add TaskList class

DanzaSeah committed:
Add Parser class

DanzaSeah committed:
Modify Ui to use TaskList class

DanzaSeah committed:
Add new constructor to TaskList

DanzaSeah committed:
Modify Storage to use TaskList

DanzaSeah committed:
Modify Parser class to handle different tasks

DanzaSeah committed:
Fix typo in Task

DanzaSeah committed:
Add showTaskList method to Ui

DanzaSeah committed:
Modify BabbleBot to be more OOP oriented

DanzaSeah committed:
Merge branch 'branch-A-MoreOOP'

DanzaSeah committed:
Divide classes into packages

DanzaSeah committed:
Merge branch 'branch-A-Packages'

DanzaSeah committed:
Add support to build and run with Gradle

DanzaSeah committed:
Merge branch 'add-gradle-support'

DanzaSeah committed:
Add support for JUnit tests on Task and TaskList

DanzaSeah committed:
Moddify build.gradle to generate jar file

DanzaSeah committed:
Write JavaDoc for BabbleBot class

DanzaSeah committed:
Add JavaDocs to the Ui class

DanzaSeah committed:
Add JavaDocs to Parser class

DanzaSeah committed:
Add JavaDocs to Storage class

DanzaSeah committed:
Add JavaDocs to Deadline class

DanzaSeah committed:
Add JavaDocs to Event class

DanzaSeah committed:
Add JavaDocs to Todo class

DanzaSeah committed:
Add JavaDocs to Task class

DanzaSeah committed:
Add JavaDocs to TaskList class

DanzaSeah committed:
Add JavaDocs to BabbleBotException class

DanzaSeah committed:
Merge branch 'branch-A-JavaDoc'

DanzaSeah committed:
Modify classes to comply with coding standard

DanzaSeah committed:
Merge branch 'branch-A-CodingStandard'

DanzaSeah committed:
Add support for "Find" with keyword

DanzaSeah committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/babblebot/parser/Parser.java
#	src/main/java/babblebot/task/Task.java

kohkakohla commented on src/main/java/babblebot/BabbleBot.java:
```diff
@@ -0,0 +1,166 @@
+     */
+    public void run() {
+        ui.sayWelcome();
+        boolean notBye = true;
```
Maybe rename this boolean to isBye to fit coding standards?

kohkakohla commented on src/test/java/babblebot/task/TaskTest.java:
```diff
@@ -0,0 +1,22 @@
+package babblebot.task;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
```
Should all the assertions imported be listed out according to the coding standards?

kohkakohla commented on src/main/java/babblebot/BabbleBot.java:
```diff
@@ -0,0 +1,166 @@
+ * It allows adding, deleting, and retrieving tasks through a TaskList.
+ */
+public class BabbleBot {
+    private static final String TASK_LIST_PATH = "./data/babblebot.txt";
```
Good naming of the constant!

kohkakohla commented on src/main/java/babblebot/parser/Parser.java:
```diff
@@ -0,0 +1,82 @@
+     */
+    public static String[] parseEventContent(String userInput) {
+        String[] result = new String[3];
+        result[0] = userInput.split("event ")[1].split(" /from")[0]; // eventContent
```
I like how you commented what each line of code does! It makes the code much more readable for me 💯

kohkakohla commented on src/main/java/babblebot/BabbleBot.java:
```diff
@@ -0,0 +1,166 @@
+        boolean notBye = true;
+
+        while (notBye) {
+            String userInp = ui.readCommand();
```
Maybe rename this to userInput? This may cause some confusion

kohkakohla reviewed (commented):
LGTM overall! Just some nits to fix.

FlyingDonkeys commented on src/main/java/babblebot/task/Task.java:
```diff
@@ -0,0 +1,86 @@
+            }
+        }
+        return false;
+
```
Its hard to find issues with your code quality, but perhaps you did not mean to add an extra newline between the return statement and the closing brace 😆

FlyingDonkeys commented on src/main/java/babblebot/ui/Ui.java:
```diff
@@ -0,0 +1,146 @@
+ */
+public class Ui {
+    private Scanner sc;
+    private String SEPARATOR = "------------------------------------------------------------------";
```
I am pretty sure this was meant to be a constant, as seen from your use of SCREAMING_SNAKE_CASE. Adding a `final` keyword to the variable might make that idea more explicit 😄

FlyingDonkeys commented on src/main/java/babblebot/ui/Ui.java:
```diff
@@ -0,0 +1,146 @@
+        System.out.println(SEPARATOR);
+        System.out.println("Whoopsie daisy! Looks like I got all tangled up in my words there!\n" +
+                "Let's try that again in a way that might make a bit more sense.\n" +
+                "What do you need help with?");
```
Good use of line breaks after the + operator to improve readability!

FlyingDonkeys reviewed (commented):
LGTM overall! Just some nits to fix.

DanzaSeah committed:
Add support for checkstyle with gradle

DanzaSeah committed:
Resolve checkstyle issues

DanzaSeah committed:
Merge branch 'branch-A-CheckStyle'

DanzaSeah committed:
Add GUI to BabbleBot

DanzaSeah committed:
Merge branch 'branch-Level-10'

DanzaSeah committed:
Add assertions to enforce task management consistency in BabbleBot

BabbleBot's current implementation lacks assertions that ensure task
management and handling behaviors remain consistent and valid. Without
these safeguards, potential programmer errors could arise from invalid
inputs or improper task manipulation, such as adding tasks without
validating input or operating on out-of-bounds indexes.

Assertions have been added to ensure:
- Non-null arguments are passed to key methods, preventing null pointers
  from causing unexpected crashes.
- TaskList consistency by verifying that tasks are correctly added and
  deleted, catching any issues where tasks may not be handled properly.
- Valid task index operations (mark, unmark, delete) by checking that
  indexes fall within bounds before performing operations.
- File operations ensure the correct saving and loading of tasks, with
  the state validated after each operation to catch any data corruption.
- Proper handling of the first interaction in the run loop, ensuring
  welcome message behavior is correctly triggered.

These assertions are implemented to catch programmer errors early during
development, ensuring the system remains robust and easier to debug in
case of future changes.

DanzaSeah committed:
Merge pull request #2 from DanzaSeah/branch-A-Assertions

Add assertions to enforce task management consistency in BabbleBot

DanzaSeah committed:
Merge branch 'master' of https://github.com/DanzaSeah/ip

* 'master' of https://github.com/DanzaSeah/ip:
  Add assertions to enforce task management consistency in BabbleBot

DanzaSeah committed:
Merge branch 'master' into branch-A-CodeQuality

DanzaSeah committed:
Refactor code for better code quality

Long sections of code were embedded within methods, reducing readability
and making the code harder to maintain. Additionally, there were several
unused methods present in the codebase, leading to unnecessary clutter.

To address these issues, the following changes were made:
- Extracted long code blocks into smaller, more focused methods.
- Removed unused methods to improve code clarity and reduce clutter.

This refactoring improves the structure and organization of the code,
making it easier to read, maintain, and extend in the future, while
ensuring that no existing functionality is affected.

DanzaSeah committed:
Merge pull request #3 from DanzaSeah/branch-A-CodeQuality

Branch a code quality

DanzaSeah committed:
Merge branch 'master' of https://github.com/DanzaSeah/ip

* 'master' of https://github.com/DanzaSeah/ip:
  Refactor code for better code quality

DanzaSeah committed:
Create TentativeEvent class

DanzaSeah committed:
Add methods for retrival and set

DanzaSeah committed:
Add saving and loading of TentativeEvents

DanzaSeah committed:
Add support for TentativeEvent scheduling

DanzaSeah committed:
Merge branch 'branch-BCD-Extension'

DanzaSeah committed:
Rectify application exit logic

DanzaSeah committed:
Change to single line imports

DanzaSeah committed:
Update README.md

DanzaSeah committed:
Update README.md

DanzaSeah committed:
Add representative screenshot Ui.png

DanzaSeah committed:
Add a user guide

DanzaSeah committed:
Merge branch 'master' of https://github.com/DanzaSeah/ip

* 'master' of https://github.com/DanzaSeah/ip:
  Update README.md
  Update README.md
